File: isUniqueChars/isUniqueChars.py
#!/usr/bin/python

from collections import defaultdict

def isUniqueChars(s):
  char_set = defaultdict(bool)
  for char in s:
    if (char_set[char]):
      return False
    char_set[char] = True
  return True

str1 = "SO NDUPE"
print str1
if (isUniqueChars(str1)):
  print "doesn't have duplicates"
else:
  print 'has duplicates'
str2 = "SO NDUPES"
print str2
if (isUniqueChars(str2)):
  print "doesn't have duplicates"
else:
  print 'has duplicates'

File: isUniqueChars/isUniqueChars.cpp
#include <iostream>
#include <string>

using std::cout;
using std::endl;
using std::string;

bool isUniqueChars(string str) {
  bool char_set[256] = {0};
  for (int i = 0; i < str.length(); i++) {
    int val = str[i];
    if (char_set[val]) {
      return false;
    }
    char_set[val] = true;
  }
  return true;
}


int main() {
  string str1 = "SO NDUPE";
  cout << str1 << endl;
  if (isUniqueChars(str1)) {
    cout << "doesn't have duplicates" << endl;
  } else {
    cout << "has duplicates" << endl;
  }
  string str2 = "SO NDUPES";
  cout << str2 << endl;
  if (isUniqueChars(str2)) {
    cout << "doesn't have duplicates" << endl;
  } else {
    cout << "has duplicates" << endl;
  }
}
